### PR TITLE
Fix the variation generation bottom sheet disappearing

### LINKED ISSUE #8193: [Variable products] "Add variation" bottom sheet is gone after device rotation
**Describe the bug**
Once `Add variation` is tapped, and the bottom sheet is presented as shown below, it will be gone after device rotation:

https://user-images.githubusercontent.com/73365754/212738556-42eca09b-0786-40cf-a49e-e031aebbfe53.mp4

**To Reproduce**
1. Open a variable product
2. Open "Variations" ~> tap "Add variation"
3. Once the bottom sheet is shown, rotate the device ~> the sheet will be gone

**Expected behavior**
The sheet stays in place after a rotation.

**Mobile Environment**
- Samsung A31
- Android 12
- WooCommerce Android version: 11.8-rc-1

### PR DESCRIPTION
**Pull request by 0nko**
Fixes #8193. Visually, it's a small change but a more substantial refactoring was required to make this work.

The PR changes the variation generation dialog into bottom sheet and utilizes the Jetpack Navigation, which automatically handles the state-saving on rotation. This resulted in the removal of a listener, which was replaced by a new bottom sheet ViewModel because the bottom sheet was used in multiple places. The tap events are handled using `navigateBackWithNotice()` calls.

[device-2023-04-28-163309.webm](https://user-images.githubusercontent.com/1522856/235177429-9ff4dff0-b74d-49c2-be17-4e9e9ad46056.webm)

**To test:**
1. Go to products
2. Tap or create a variable product
3. Tap on Add variations
4. (add new attributes and options, if there are none)
5. Tap on the Add variation button
6. Notice a bottom sheet dialog appears
7. Rotate the screen
8. Notice the bottom sheet doesn't disappear
9. Tap on Generate all variations
10. Notice the generation dialog appears
11. Tap cancel
12. Tap on Add variation button again
13. Notice the bottom sheet appears
14. Tap on Add new variation
15. Notice a variation is generated

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/VariationListFragment.kt [hunk 1]
```diff
@@ -285,10 +284,9 @@ class VariationListFragment :
     }
 
     private fun showAddVariationSelectDialog() {
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/VariationListFragment.kt]: Np, maybe this function could also be renamed accordingly. We are displaying a bottom sheet now.

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/products/variations/GenerateVariationBottomSheetViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import org.mockito.kotlin.verify
+
+@ExperimentalCoroutinesApi
+class GenerateVariationBottomSheetViewModelTest : BaseUnitTest() {
+    private lateinit var viewModel: GenerateVariationBottomSheetViewModel
+    private val savedStateHandle: SavedStateHandle = mock()
+    private val eventObserver: Observer<Event> = mock()
+
+    @get:Rule
+    var instantTaskExecutorRule = InstantTaskExecutorRule()
```
Comment (JorgeMucientes) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/products/variations/GenerateVariationBottomSheetViewModelTest.kt]: hmm, curious why do you need this? `BaseUnitTest` already sets an `InstantTaskExecutorRule`

**Review by JorgeMucientes — APPROVED** [review]

**Comment by JorgeMucientes** [comment]
Good work @0nko! Nice refactor and code works as expected. I left a question and a minor nitpick. Nothing blocking.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/VariationListFragment.kt (hunk 1 above)
Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/variations/VariationListFragment.kt]: Good call 👍

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/products/variations/GenerateVariationBottomSheetViewModelTest.kt (hunk 2 above)
Comment (0nko) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/products/variations/GenerateVariationBottomSheetViewModelTest.kt]: Ha, ask ChatGPT! 😆 I missed this one.

**Pull request merged**